tapir/coop/services/MembershipResignationService.py

```diff
@@ -0,0 +1,89 @@
+import datetime
+
+from django.db import transaction
+from tapir.accounts.models import TapirUser
+from tapir.coop.models import MembershipResignation, ShareOwnership
+from tapir.shifts.models import (
+    ShiftAttendanceTemplate,
+    ShiftAttendance,
+    ShiftAccountEntry,
+)
+from tapir.shifts.services.frozen_status_service import FrozenStatusService
+from tapir.shifts.models import ShiftUserData, ShiftAttendanceMode
+
+from tapir.utils.shortcuts import get_timezone_aware_datetime
+
+
+class MembershipResignationService:
+    @staticmethod
+    @transaction.atomic
+    def update_shifts_and_shares(self, resignation: MembershipResignation):
+        tapir_user: TapirUser = getattr(resignation.share_owner, "user", None)
+        if not tapir_user:
+            print("Couldn't find an existing Tapir user.")
```

Comment:
Shouldn't this be an exception?
Or if it's an expected case then we don't need the print.

Reply:
Deleted